modules/mqtt/destination/mqtt-worker.c

```diff
@@ -191,9 +191,11 @@ _connect(LogThreadedDestWorker *s)
   gint rc;
 
   MQTTClient_connectOptions conn_opts = MQTTClient_connectOptions_initializer;
-
   conn_opts.keepAliveInterval = owner->keepalive;
   conn_opts.cleansession = FALSE;
+  conn_opts.username = owner->username;
+  conn_opts.password = owner->password;
```

Comment:
Is it okay to use these unconditionally?
For example can it cause a problem if only password is given?

If yes, then it would be better to notify the user about their incomplete config at syslog-ng init time, rather than at connect time.

Reply:
NULL/0-values are considered as unset fields in this struct.
This is documented (it is not just an implementation detail):

https://www.eclipse.org/paho/files/mqttdoc/MQTTClient/html/struct_m_q_t_t_client__connect_options.html